frontend/include/chpl/uast/AstNode.h

```diff
@@ -57,6 +57,7 @@ class AstNode {
  private:
   AstTag tag_;
   ID id_;
+  int attributeGroupChildNum_;
```

Comment:
Can you store this field immediately after AstTag? That will optimize the memory a bit & that's worthwhile because this applies to all ast nodes.